okhttp/src/test/java/okhttp3/HttpUrlTest.java

```diff
@@ -440,6 +440,13 @@ HttpUrl parse(String url) {
         .test(Component.HOST);
   }
 
+  /** This one's ugly: the HttpUrl's host is non-empty, but the URI's host is null. */
+  @Test public void hostContainsOnlyStrippedCharacters() throws Exception {
+    HttpUrl url = parse("http://>/");
+    assertThat(url.host()).isEqualTo(">");
```

Comment:
Should we add validation in `HttpUrl` that prevents the user from having an _invalid_ hostname?

Reply:
Yes, but...

We’d need to define what a valid hostname is. That’s difficult, and lots of libraries that have attempted this have failed.